CONTRIBUTING.md

```diff
@@ -0,0 +1,81 @@
+# Contributing to Yarn
+
+Yarn is in a slightly precarious position, due to the odd nature of MCP. MCP is all-rights-reserved, yet visible to anyone
+who has ever modded Minecraft using anything other than Fabric. This leads to "contamination" of a lot of potential
+contributors, since people already have a conception of what a method "should" be named. Because of this, we have decided to
+take a clean-room approach with mapping Minecraft to avoid potential legal/drama issues within the community.
+
+## What's in a Name?
+
+The fundamental goal of mappings is to make modding Minecraft easier to understand, for beginners and experts alike. We want
+to create a cohesive, simple naming scheme that avoids confusion or being too verbose. Having a lot of things named in
+different schemes would be a nightmare for usability, so we want to avoid that. This includes naming schemes not just within
+Yarn but within the greater Minecraft community. Despite that goal, there are some name sources that are inherently more
+"reliable" than others; you should use an up-to-date and known Mojang name above any alternative, even if an alternative
```

Comment:
"you should use an up-to-date and known Mojang name [...] be simpler." -> "if an official Mojang name is hinted at or can be guessed, you should generally use that - sticking closer to official terminology helps everyone."